odl/space/pspace.py

```diff
@@ -390,6 +391,14 @@ def element(self, inp=None, cast=True):
 
         return self.element_type(self, parts)
 
+    @property
+    def examples(self):
+        """Return examples from all sub-spaces."""
+        for examples in product(*[spc.examples for spc in self.spaces]):
+            name = ', '.join(name for name, _ in examples)
+            vector = self.element([vec for _, vec in examples])
+            yield (name, vector)
```

Comment:
Not quite sure how to skip x elements in iterators, also we dont know how many exist in the subspaces. This would be non-trivial to say the least. I agree with the sentiment though.